firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/metadata/RolloutAssignment.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.metadata;
+
+import com.google.auto.value.AutoValue;
+import com.google.firebase.encoders.DataEncoder;
+import com.google.firebase.encoders.annotations.Encodable;
+import com.google.firebase.encoders.json.JsonDataEncoderBuilder;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * model used in user metadata context which make rollout assignment serialize, deserialize easier
+ */
+@Encodable
+@AutoValue
+public abstract class RolloutAssignment {
+
+  @Encodable.Field(name = "rolloutId")
+  public abstract String rolloutId();
```

Comment:
Why do you need to set the field name manually, isn't it able to get the name automatically from the method name? Do you need to name them like `getRolloutId()`?

Reply:
Ah I see, let me change to getRolloutId(), I was following some example within the codebase.